File: benchmark-cpp/src/filter.cpp
#include "filter.h"
#include "column.h"

template<typename T>
Filter<T>::Filter(T value, FilterOp op, Column col) {
    this->cmp_val = value;
    this->op = op;
    this->col = col;
}

template <typename T>
bool Filter<T>::execute(T val)
{
    switch (op) {
    case FilterOp::Gt:
        return val > cmp_val;
    case FilterOp::Lt:
        return val < cmp_val; 
    case FilterOp::Eq:
        return (val == cmp_val);
    case FilterOp::Gte:
        return val >= cmp_val;
    case FilterOp::Lte:
        return val <= cmp_val;
    }
}

template <typename T>
Column Filter<T>::get_column()
{
    return col;
}

File: benchmark_python/src/core/filter.py
from enum import Enum

class Filter:
    def __init__(self, col, op, val):
        self.cmp_val = val
        self.op = op
        self.col = col

    def execute(self, val): 
        match self.op:
            case FilterOp.EQ:
                return (val == self.cmp_val)
            case FilterOp.GT:
                return (val > self.cmp_val)
            case FilterOp.GTE:
                return (val >= self.cmp_val)
            case FilterOp.LT:
                return (val < self.cmp_val)
            case FilterOp.LTE:
                return (val <= self.cmp_val)

class FilterOp(Enum):
    EQ  = 1
    GT  = 2
    GTE = 3
    LT  = 4 
    LTE = 5